pkg/scheduler/schedule_one.go

```diff
@@ -99,9 +111,15 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	}
 
 	// bind the pod to its host asynchronously (we can do this b/c of the assumption step above).
+	//
+	// This goroutine becomes responsible for calling DonePod. assumedPodInfo returns a different
+	// pointer, so we have to remember the original one here.
+	donePodInfo := podInfo
+	podInfo = nil
```

Comment:
The exact pointer isn't needed anymore with the final approach. Should DonePod aka PriorityQueue.Done perhaps be changed to taking a UID as parameter?

Reply:
> Should DonePod aka PriorityQueue.Done perhaps be changed to taking a UID as parameter?

Yes go ahead, that sounds cleaner.

Reply:
Done.